test/testcafe/spec/ChallengeAuthenticatorEmail_spec.js

```diff
@@ -214,17 +214,44 @@ test
   });
 
 test
-  .requestHooks(logger, invalidOTPMockContinuePoll)('continue polling on form error', async t => {
+  .requestHooks(logger, invalidOTPMockContinuePoll)('continue polling on form error with dynamic polling', async t => {
     const challengeEmailPageObject = await setup(t);
+    await t.expect(challengeEmailPageObject.resendEmailView().hasClass('hide')).ok();
+
+    // 2 poll requests in 2 seconds at 1 sec interval
+    await t.wait(2000);
+    await t.expect(logger.count(
+      record => record.response.statusCode === 200 &&
+        record.request.url.match(/poll/)
+    )).eql(2);
+
+    await t.removeRequestHooks(dynamicRefreshShortIntervalMock);
+    await t.addRequestHooks(dynamicRefreshLongIntervalMock);
+
+    // 1 poll requests in 2 seconds at 2 sec interval
+    await t.wait(2000);
+    await t.expect(logger.count(
+      record => record.response.statusCode === 200 &&
+        record.request.url.match(/poll/)
+    )).eql(3);
+
+    await t.removeRequestHooks(dynamicRefreshLongIntervalMock);
+    await t.addRequestHooks(invalidOTPMock);
+
     await challengeEmailPageObject.verifyFactor('credentials.passcode', 'xyz');
     await challengeEmailPageObject.clickNextButton();
     await challengeEmailPageObject.waitForErrorBox();
     await t.expect(challengeEmailPageObject.getInvalidOTPError()).contains('Authentication failed');
-    await t.wait(5000);
+
+    await t.removeRequestHooks(invalidOTPMock);
+    await t.addRequestHooks(dynamicRefreshLongIntervalMock);
+
+    // 2 poll requests in 4 seconds at 2 sec interval
+    await t.wait(4000);
     await t.expect(logger.count(
       record => record.response.statusCode === 200 &&
       record.request.url.match(/poll/)
-    )).eql(2);
+    )).eql(5);
```

Comment:
if 2 poll request, then it should be 2 right ? as per comments ?

Reply:
It's cumulative 2 (first 2 requests in 2 seconds with 1 second refresh)+ 1 (1 request in 2 second with 2 second refresh) + 2 (2 requests in 4 second with 2 second refresh) = 5 requests total over the test scenario

Reply:
Cool.. can you mention that in comment as well.

Reply:
sure